### ISSUE TITLE
hidden attribute gets set to yes on [unit] creation

### ISSUE THREAD

**Issue by virtualghetto** [wesnoth/wesnoth#4889]
### Game and System Information
Wesnoth 1.14.9
Debian sid package

### Describe the bug
Under certain conditions, when a unit that has `[hides]` ability is placed on the map with `[unit]`, it's `hidden` flag gets set to yes and doesn't get reset automatically.
 
Details are in this thread:
https://forums.wesnoth.org/viewtopic.php?f=21&t=52446

### To Reproduce
Create a Shadow at night or an Elvish Avenger on a forest tile with `[unit]`

**Comment by soliton-** [comment]
So when the night is over or you move the Avenger out of the forest it stays hidden?

**Comment by virtualghetto** [comment]
Yes, I attached a replay on the forum thread of it happening with both the Avenger and the Shadows during the day.
This might be related to having animate=yes on creation.

**Comment by virtualghetto** [comment]
To be clear, this issue affect the ai controlled units, not the human controlled ones.
